src/base/memory_monitor_sr.cpp

```diff
@@ -20,21 +20,67 @@
  * memory_monitor_sr.cpp - Implementation of memory monitor module
  */
 
+#include <malloc.h>
+#include <cassert>
 #include <cstring>
 #include <algorithm>
-#include <cassert>
 
 #include "memory_monitor_sr.hpp"
 
+#ifndef HAVE_USR_INCLUDE_MALLOC_H
+#define HAVE_USR_INCLUDE_MALLOC_H
+#endif
+
+typedef struct mmon_metainfo MMON_METAINFO;
+struct mmon_metainfo
+{
+  uint64_t allocated_size;
+  int tag_id;
+  int magic_number;
+};
+
 namespace cubmem
 {
+  memory_monitor *mmon_Gl = nullptr;
+
   memory_monitor::memory_monitor (const char *server_name)
-    : m_server_name {server_name} {}
+    : m_server_name {server_name},
+      m_magic_number {*reinterpret_cast <const int *> ("MMON")}
+  {}
 
-  static std::string memory_monitor::make_tag_name (const char *file, const int line)
+  size_t memory_monitor::get_allocated_size (const char *ptr)
+  {
+#if defined(WINDOWS)
+    size_t allocated_size = _msize ((void *)ptr);
```

Comment:
I heard that memory monitoring does not support WINDOWS platform. 
Has this constraint been lifted ?

Reply:
This change is for success window build, not for support WINDOWS. This topic seems to require a discussion on safety aspects possibly after all implementations are completed.

Reply:
Rather than handling each case, why not exclude all in Windows?